tests/unit/lms/tasks/email_digests_test.py

```diff
@@ -192,10 +199,66 @@ def test_it_crashes_if_updated_after_or_updated_before_is_invalid(
         self, updated_after, updated_before
     ):
         with pytest.raises(ValueError, match="^Invalid isoformat string"):
-            send_instructor_email_digests(
-                sentinel.h_userids, updated_after, updated_before
+            send_instructor_email_digests(  # pylint:disable=no-value-for-parameter
+                h_userids=sentinel.h_userids,
+                updated_after=updated_after,
+                updated_before=updated_before,
             )
 
+    def test_it_retries_if_sending_fails(self, digest_service, retry):
+        updated_after = datetime(year=2023, month=3, day=1)
+        updated_before = datetime(year=2023, month=3, day=2)
+        # send_instructor_email_digests() raises a SendDigestsError whose
+        # errors attribute is a dict mapping the h_userid's that failed to send
+        # to their corresponding MailchimpError's.
+        digest_service.send_instructor_email_digests.side_effect = SendDigestsError(
+            errors={
+                sentinel.failed_h_userid_1: MailchimpError(),
+                sentinel.failed_h_userid_2: MailchimpError(),
+            }
+        )
+
+        send_instructor_email_digests(  # pylint:disable=no-value-for-parameter
+            h_userids=sentinel.h_userids,
+            updated_after=updated_after.isoformat(),
+            updated_before=updated_before.isoformat(),
+            override_to_email=sentinel.override_to_email,
+        )
+
+        # It retries the task with the same arguments except that h_userids is
+        # now only those userids that failed to send.
+        retry.assert_called_once_with(
+            kwargs={
+                "h_userids": Any.list.containing(
+                    [sentinel.failed_h_userid_1, sentinel.failed_h_userid_2]
+                ).only(),
+                "updated_after": updated_after.isoformat(),
+                "updated_before": updated_before.isoformat(),
+                "override_to_email": sentinel.override_to_email,
+            },
+            countdown=Any.int(),
+        )
+
+    def test_it_retries_if_sending_fails_with_an_unexpected_exception(
+        self, digest_service, retry
+    ):
+        digest_service.send_instructor_email_digests.side_effect = RuntimeError(
+            "Unexpected"
+        )
+
+        send_instructor_email_digests(  # pylint:disable=no-value-for-parameter
+            h_userids=sentinel.h_userids,
+            updated_after="2023-02-27T00:00:00",
+            updated_before="2023-02-28T00:00:00",
+            override_to_email=sentinel.override_to_email,
+        )
+
+        retry.assert_called_once_with(countdown=Any.int())
+
+    @pytest.fixture
+    def retry(self, patch):
+        return patch("lms.tasks.email_digests.send_instructor_email_digests.retry")
```

Comment:
Mocking just the `retry()` method as [recommended in Celery's docs](https://docs.celeryq.dev/en/stable/userguide/testing.html#tasks-and-unit-tests)